openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/RelatedTerms.tsx

```diff
@@ -77,14 +78,12 @@ const RelatedTerms = ({
   const suggestionSearch = (searchText = '') => {
     setIsLoading(true);
     searchData(searchText, 1, PAGE_SIZE, '', '', '', SearchIndex.GLOSSARY)
-      .then((res: SearchResponse) => {
-        const termResult = (
-          formatSearchGlossaryTermResponse(
-            res?.data?.hits?.hits || []
-          ) as EntityReference[]
+      .then((res) => {
+        const termResult = formatSearchGlossaryTermResponse(
+          (res?.data as SearchResponse<SearchIndex.GLOSSARY>)?.hits?.hits
         ).filter((item) => {
           return item.fullyQualifiedName !== glossaryTerm.fullyQualifiedName;
-        });
+        }) as unknown as EntityReference[];
```

Comment:
I would recommend transforming the data in the shape of EntityReference instead of type casting it.

Reply:
Done